backend/tests/gameEndpointTests.ml

```diff
@@ -183,31 +188,74 @@ let tests = [
                 player_one = Some DomainTests.another_minimal_user
             } in
             FirestoreTests.Mock.Game.set game;
 
             (* When resigning from it *)
             let target = Printf.sprintf "game/%s?action=resign" game_id in
-            let request = Dream.request ~method_:`POST ~target "" in
+            let request : Dream.request = Dream.request ~method_:`POST ~target "" in
             let* result = handler request in
 
             (* Then it should resign from the game *)
             check status "response status" `OK (Dream.status result);
-            let winner = DomainTests.another_minimal_user in
-            let loser = DomainTests.a_minimal_user in
-            let end_game = Game.Updates.End.(to_yojson (get ~winner ~loser Game.GameResult.Resign)) in
+            let winner : MinimalUser.t = DomainTests.another_minimal_user in
+            let loser : MinimalUser.t = DomainTests.a_minimal_user in
+            let end_game : Yojson.Safe.t = Game.Updates.End.to_yojson (Game.Updates.End.get ~winner ~loser Game.GameResult.Resign) in
             let end_game_event = GameEvent.(to_yojson (Action (Action.end_game DomainTests.a_minimal_user (now * 1000)))) in
+            let first_game_is_lost : JSON.t = Domain.User.EloInfo.to_yojson DomainTests.first_game_is_lost in
+            let first_game_is_won : JSON.t = Domain.User.EloInfo.to_yojson DomainTests.first_game_is_won in
             let expected = [
                 FirestoreTests.UpdateGame (game_id, end_game);
                 FirestoreTests.AddEvent (game_id, end_game_event);
+                FirestoreTests.UpdateElo (loser.id, type_game, first_game_is_lost);
+                FirestoreTests.UpdateElo (winner.id, type_game, first_game_is_won);
+            ] in
+            check (list FirestoreTests.call) "calls" expected !FirestoreTests.Mock.calls;
+            Lwt.return ()
+        );
+
+        lwt_test "should update elo when resigning" (fun () ->
+            FirestoreTests.Mock.clear_calls ();
+            AuthTests.Mock.set DomainTests.a_minimal_user.id DomainTests.a_user;
+            let now : int = 42 in
+            let type_game : string = "P4" in
+            ExternalTests.Mock.current_time_seconds := now;
+
+            (* Given a game with an opponent *)
+            let game_id : string = "game_id" in
+            let game : Domain.Game.t = {
+                (Game.initial type_game DomainTests.a_minimal_user DomainTests.a_minimal_user_current_elo) with
+                player_one = Some DomainTests.another_minimal_user
+            } in
+            FirestoreTests.Mock.Game.set game;
+
+            (* When resigning from it *)
+            let target = Printf.sprintf "game/%s?action=resign" game_id in
+            let request : Dream.request = Dream.request ~method_:`POST ~target "" in
+            let* result = handler request in
+
+            (* Then it should resign from the game *)
```

Comment:
mauvais then

Reply:
ah ? tu proposes quoi ?

Reply:
bah le test c'est "should update elo"
"then it should resign" n'implique aucunement que l'elo est updaté !

Reply:
bon c'était un test copié/collé sans modifications, donc j'lai buté et adapté le GWT de l'original

Reply:
c'pas sans modif si ?

Reply:
wablief? Le copier coller était sans modif oui (enfin, les même modif dans l'original & le collé)

Reply:
le fameux "copier/coller" était à quel ligne dans quel commit ?

Reply:
le commit du commentaire original (98fd9468ef95f4d0) de ce fichier, ligne 227

Reply:
Je sais pas, sur ce commit y a pas de commentaires à cette ligne là, mais bon, TPR